frontend/catalyst/api_extensions/callbacks.py

```diff
@@ -38,6 +40,64 @@
 from catalyst.utils.types import convert_pytype_to_shaped_array
 
 
+def accelerate(func=None, dev=None):
+    """Seamless integration with jax.jit
+
+    The accelerate function can be used to automatically apply `jax.jit` to
+    the input function `func` and be used to specify the device with the `dev`
+    keyword argument. If `dev` is None, then the default device as determined
+    by JAX is used.
+
+    Can be used in the following modes:
+
+
+    ```py
+    @accelerate
+    def identity(x):
+      return x
+
+    @qml.qjit
+    def qjitted_fn(x):
+      return x
+    ```
+
+    ```
+    @accelerate(dev=jax.devices()[0])
+    def identity(x):
+      return x
+
+    @qml.qjit
+    def qjitted_fn(x):
+      return x
+    ```
+
+    or
+
+    ```
+    @qml.qjit
+    def qjitted_fn(x):
+      return accelerate(lambda x: x)
+    ```
+    """
+
+    if isinstance(func, Callable):
+
+        @functools.wraps(func)
+        def defer(*args, **kwargs):
+            # Make abstract variables from input tracers.
+            absargs, abskwargs = tree_map(shaped_abstractify, (args, kwargs))
+            # Find the shape of the return value
+            _, returnshape = jax.make_jaxpr(func, return_shape=True)(*absargs, **abskwargs)
+            jitted_fn = jax.jit(func, device=dev)
```

Comment:
note that `jax.jit(..., device=...)` is deprecated:

https://github.com/google/jax/blob/25a47649d2bf5563708f11dfbd15b187287773f7/jax/_src/pjit.py#L407-L411

Instead, they direct users to use `jax.device_put` on all arguments to the function.

Reply:
I've changed it here: https://github.com/PennyLaneAI/catalyst/pull/805/commits/ba899a3473fcafcc4e37b56aea13522a07349db4

Now all arguments before calling the `jax.jit`ed function will be moved using `jax.device_put`